polkadot/runtime/parachains/src/coretime/migration.rs

```diff
@@ -130,8 +138,71 @@ pub fn migrate_to_coretime<T: Config>() -> Weight {
 		c.coretime_cores = total_cores;
 	});
 
+	if let Err(err) = migrate_send_assignments_to_coretime_chain::<T>() {
+		log::error!("Sending legacy chain data to coretime chain failed: {:?}", err);
+	}
+
 	let single_weight = <T as Config>::WeightInfo::assign_core(1);
 	single_weight
 		.saturating_mul(u64::from(legacy_count.saturating_add(config.coretime_cores)))
 		.saturating_add(T::DbWeight::get().reads_writes(1, 1))
 }
+
+fn migrate_send_assignments_to_coretime_chain<T: Config>() -> result::Result<(), SendError> {
+	let legacy_paras = paras::Pallet::<T>::parachains();
+	let (system_chains, lease_holding): (Vec<_>, Vec<_>) =
+		legacy_paras.into_iter().partition(IsSystem::is_system);
+
+	let reservations = system_chains.into_iter().map(|p| {
+		let schedule = BoundedVec::truncate_from(vec![ScheduleItem {
+			mask: CoreMask::complete(),
+			assignment: CoreAssignment::Task(p.into()),
+		}]);
+		mk_coretime_call(super::CoretimeCalls::Reserve(schedule))
+	});
+
+	let leases = lease_holding.into_iter().filter_map(|p| {
+		let Some(valid_until) = T::GetLegacyLease::get_parachain_lease_in_blocks(p) else {
+			log::error!("Lease holding chain with no lease information?!");
+			return None
+		};
+		let valid_until: u32 = match valid_until.try_into() {
+			Ok(val) => val,
+			Err(_) => {
+				log::error!("Converting block number to u32 failed!");
+				return None
+			},
+		};
+		// We assume the coretime chain set this parameter to the recommened value in RFC-1:
+		const TIME_SLICE_PERIOD: u32 = 80;
+		let round_up = if valid_until % TIME_SLICE_PERIOD > 0 { 1 } else { 0 };
+		let time_slice = valid_until / TIME_SLICE_PERIOD + TIME_SLICE_PERIOD * round_up;
+		Some(mk_coretime_call(super::CoretimeCalls::SetLease(p.into(), time_slice)))
+	});
+
+	let message_content = iter::once(Instruction::UnpaidExecution {
+		weight_limit: WeightLimit::Unlimited,
+		check_origin: None,
+	})
+	.chain(reservations)
+	.chain(leases)
+	.collect();
+
+	let message = Xcm(message_content);
+	send_xcm::<T::SendXcm>(
+		MultiLocation {
+			parents: 0,
```

Comment:
```suggestion
			parents: 1,
```

Where is this `migrate_send_assignments_to_coretime_chain` deployed or called? on coretime parachain?
If so, then `parents: 0` needs to be changed to `parents: 1`

Reply:
ok, I see bellow, its on rococo, so `parents: 0` is ok

Reply:
No it is called on the relay chain.

Reply:
No on the relay chain itself. (It is a dmp to the coretime chain to bring it up to the relay chain configuration)